filebeat/docs/modules/kafka.asciidoc

```diff
@@ -3,30 +3,78 @@ This file is generated! See scripts/docs_collector.py
 ////
 
 [[filebeat-module-kafka]]
+:modulename: kafka
+
 == Kafka module
 
-This module collects and parses the logs created by https://kafka.apache.org/[Kafka].
+The +{modulename}+ module collects and parses the logs created by
+https://kafka.apache.org/[Kafka].
+
+include::../include/what-happens.asciidoc[]
 
 [float]
 === Compatibility
 
-The Kafka module was tested with logs from versions 2.11.
+The +{modulename}+ module was tested with logs from versions 2.11.
+
+include::../include/running-modules.asciidoc[]
 
 [float]
-=== Dashboard
+=== Example dashboard
 
-This module comes with a sample dashboard to see Kafka logs and stacktraces.
+This module comes with a sample dashboard to see Kafka logs and stack traces.
 
+[role="screenshot"]
 image::./images/filebeat-kafka-logs-overview.png[]
 
-[float]
-=== Logs fileset settings
+include::../include/configuring-intro.asciidoc[]
+
+The following example shows how to set paths in the +modules.d/{modulename}.yml+
+file to override the default paths for logs:
+
+["source","yaml",subs="attributes"]
+-----
+- module: kafka
+  log:
+    enabled: true
+    var.paths:
+      - "/path/to/logs/controller.log*"
+      - "/path/to/logs/server.log*"
+      - "/path/to/logs/state-change.log*"
+      - "/path/to/logs/kafka-*.log*"
+-----
+
+
+// REVIEWERS: I must be doing something wrong with the config settings. The
+// above config works, but when I try to specify var.kafka_home, it doesn't
+// seem to have any effect. 
```

Comment:
If you don't overwrite `var.paths` then it can be used to change the base dir on all the default paths.

I think it will only have an effect if it is used in one of the paths you have specified like `"{{.kafka_home}}/logs/kafka-*.log*"`.

Reply:
@andrewkroh Yeah, that's what I thought, too, but I couldn't get it to work. Can you see if I'm doing something wrong?

Here's what the modules.d/kafka.yml file contains:

````
- module: kafka
  # All logs
  log:
    enabled: true

    # Set custom paths for Kafka. If left empty,
    # Filebeat will look under /opt.
    var.kafka_home: /Users/dedemorton/test_kafka*

    # Set custom paths for the log files. If left empty,
    # Filebeat will choose the paths depending on your OS.
    #var.paths:
````

My log files for testing are here: 

![image](https://user-images.githubusercontent.com/14206422/34059675-c22977d8-e194-11e7-84c8-a02672aaa631.png)

The manifest.yml file hasn't changed: 

````
module_version: 1.0

var:
  - name: kafka_home
    default: /opt/kafka*
  - name: paths
    default:
      - "{{.kafka_home}}/logs/controller.log*"
      - "{{.kafka_home}}/logs/server.log*"
      - "{{.kafka_home}}/logs/state-change.log*"
      - "{{.kafka_home}}/logs/kafka-*.log*"

ingest_pipeline: ingest/pipeline.json
prospector: config/log.yml 

````

When I run ./filebeat -e, I see:

````
Rhodas-MacBook-Pro:filebeat-6.1.0-darwin-x86_64 dedemorton$ ./filebeat -e
2017/12/15 20:44:05.578439 metrics.go:23: INFO Metrics logging every 30s
2017/12/15 20:44:05.578435 beat.go:436: INFO Home path: [/Users/dedemorton/BuildTesting/6.1.0_0c6d5083/filebeat-6.1.0-darwin-x86_64] Config path: [/Users/dedemorton/BuildTesting/6.1.0_0c6d5083/filebeat-6.1.0-darwin-x86_64] Data path: [/Users/dedemorton/BuildTesting/6.1.0_0c6d5083/filebeat-6.1.0-darwin-x86_64/data] Logs path: [/Users/dedemorton/BuildTesting/6.1.0_0c6d5083/filebeat-6.1.0-darwin-x86_64/logs]
2017/12/15 20:44:05.578703 beat.go:443: INFO Beat UUID: 018a0614-b0cf-4543-84b5-0e17dd774a1b
2017/12/15 20:44:05.578721 beat.go:203: INFO Setup Beat: filebeat; Version: 6.1.0
2017/12/15 20:44:05.578958 client.go:123: INFO Elasticsearch url: http://localhost:9200
2017/12/15 20:44:05.579242 module.go:76: INFO Beat name: Rhodas-MacBook-Pro.local
2017/12/15 20:44:05.579722 beat.go:276: INFO filebeat start running.
2017/12/15 20:44:05.579765 registrar.go:71: INFO No registry file found under: /Users/dedemorton/BuildTesting/6.1.0_0c6d5083/filebeat-6.1.0-darwin-x86_64/data/registry. Creating a new registry file.
2017/12/15 20:44:05.580178 registrar.go:108: INFO Loading registrar data from /Users/dedemorton/BuildTesting/6.1.0_0c6d5083/filebeat-6.1.0-darwin-x86_64/data/registry
2017/12/15 20:44:05.580207 registrar.go:119: INFO States Loaded from registrar: 0
2017/12/15 20:44:05.580222 crawler.go:48: INFO Loading Prospectors: 1
2017/12/15 20:44:05.580253 registrar.go:150: INFO Starting Registrar
2017/12/15 20:44:05.582697 crawler.go:82: INFO Loading and starting Prospectors completed. Enabled prospectors: 0
2017/12/15 20:44:05.582727 reload.go:127: INFO Config reloader started
2017/12/15 20:44:05.583537 reload.go:258: INFO Starting 1 runners ...
2017/12/15 20:44:05.583582 client.go:123: INFO Elasticsearch url: http://localhost:9200
2017/12/15 20:44:05.604042 client.go:651: INFO Connected to Elasticsearch version 6.1.0
2017/12/15 20:44:05.608167 prospector.go:87: INFO Starting prospector of type: log; ID: 13205974736160889200 
2017/12/15 20:44:05.608198 reload.go:219: INFO Loading of config files completed.
2017/12/15 20:44:35.579798 metrics.go:39: INFO Non-zero metrics in the last 30s: beat.info.uptime.ms=30004 beat.memstats.gc_next=4194304 beat.memstats.memory_alloc=1886704 beat.memstats.memory_total=3403792 filebeat.harvester.open_files=0 filebeat.harvester.running=0 libbeat.config.module.running=1 libbeat.config.module.starts=1 libbeat.config.reloads=1 libbeat.output.type=elasticsearch libbeat.pipeline.clients=2 libbeat.pipeline.events.active=0 registrar.states.current=0 registrar.writes=1
2017/12/15 20:45:05.578798 metrics.go:39: INFO Non-zero metrics in the last 30s: beat.info.uptime.ms=30000 beat.memstats.gc_next=4194304 beat.memstats.memory_alloc=1980384 beat.memstats.memory_total=3497472 filebeat.harvester.open_files=0 filebeat.harvester.running=0 libbeat.config.module.running=1 libbeat.pipeline.clients=2 libbeat.pipeline.events.active=0 registrar.states.current=0
````

Reply:
Probably @tsg would be the best person help here. I don't see an indication that the module is running in that log output. `"Enabled prospectors: 0"`